File: segundo/1er cuatri/PDOO/Irrgarten_java/src/irrgarten/CombatElement.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author alvarogz
 */
public abstract class CombatElement {
    private float effect;
    private int uses;
    
    public CombatElement(float effect, int uses) {
        this.effect = effect;
        this.uses = uses;
    }
    
    protected float produceEffect(){
        if (uses > 0){
            uses--;
            return effect;
        }
        else
            return 0;
    }
    
    public boolean discard() {
        return Dice.discardElement(uses);
    }
    
    @Override
    public String toString(){
        return this.effect + ", " + this.uses;
    }
}

File: segundo/1er cuatri/PDOO/irrgarten_ruby/Irrgarten/combat_element.rb
# frozen_string_literal: true
require_relative 'dice'

class CombatElement
  def initialize(effect, uses)
    @effect = effect
    @uses = uses
  end

  def produce_effect()
    if @uses > 0
      @uses-=1
      @effect
    else
      0
    end
  end

  def discard()
    Dice.discard_element(@uses)
  end

  def to_s
    "#{@effect}, #{@uses}"
  end

  protected :produce_effect
end
